core/src/main/java/io/confluent/rest/ApplicationServer.java

```diff
@@ -95,10 +100,32 @@ static boolean isJava11Compatible() {
   }
 
   public ApplicationServer(T config) {
-    this(config, createThreadPool(config));
+    this(config, createThreadPool(config), null);
+  }
+
+  private X509Source initializeSpiffeX509Source(RestConfig config) {
+    if (!config.getBoolean(RestConfig.SSL_IS_SPIRE_ENABLED_CONFIG)) {
+      return null;
+    }
+
+    try {
+      String spiffeSocketPath = config.getString(RestConfig.SSL_SPIRE_AGENT_SOCKET_PATH_CONFIG);
+      if (spiffeSocketPath == null || spiffeSocketPath.isEmpty()) {
+        throw new Exception("spiffeSocketPath is required when SPIFFE is enabled, but it is empty; please specify the path in the config");
+      }
+
+      DefaultX509Source.X509SourceOptions x509SourceOptions = DefaultX509Source.X509SourceOptions
+              .builder()
+              .spiffeSocketPath(spiffeSocketPath)
+              .svidPicker(list -> list.get(list.size() - 1))
+              .build();
+      return DefaultX509Source.newSource(x509SourceOptions);
+    } catch (Exception e) {
+      throw new RuntimeException("Failed to initialize SPIFFE X509 source", e);
+    }
```

Comment:
if we initialize in applicationServer, we have to think about multiple apps ; how will they interact/use this X509 Source? Or should we move to app level?